app/main/views/frameworks.py

```diff
@@ -540,12 +535,24 @@ def upload_framework_agreement(framework_slug):
         ), 400
 
     agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
-    legal_supplier_name = supplier_framework['declaration']['SQ1-1a']
     extension = get_extension(request.files['agreement'].filename)
-    path = get_agreement_document_path(
-        framework_slug, current_user.supplier_id, legal_supplier_name,
-        'signed-framework-agreement{}'.format(extension))
-    agreements_bucket.save(path, request.files['agreement'], acl='private')
+
+    # TODO: get rid of this; rewrite `.get_agreement_document_path()` function in dmutils
+    path = '{0}/agreements/{1}/{1}-signed-framework-agreement{2}'.format(
+        framework_slug,
+        current_user.supplier_id,
+        extension
+    )
+    agreements_bucket.save(
+        path,
+        request.files['agreement'],
+        acl='private',
+        download_filename='{}-{}-signed-framework-agreement{}'.format(
```

Comment:
We should be aware that if the supplier name is changed the agreements files will still be downloaded with the previous supplier name. That may be ok as it's the supplier name at the point that the agreement was uploaded.